### PR TITLE
Improve app security around app integrations

### PR DESCRIPTION
**Pull request by dcramer**
- Adds some basic branch coverage for a created, but uninstalled integration
- Require security reviewers for auth and permissions paths

### CODE REVIEW

**Comment by dcramer** [comment]
Looks like that wasn't enough to trigger it. Have some free time this morning so I'll try to expand.

**Comment by dcramer** [comment]
Confirm the `test_no_access_due_to_invalid_user` test covers the case.

File: tests/sentry/auth/test_access.py [hunk 1]
```diff
@@ -225,8 +229,22 @@ def test_has_access(self):
...
+        assert not result.has_team_access(self.team2)
+        assert not result.has_team_access(self.out_of_scope_team)
+
+    def test_no_access_due_to_no_installation_owned(self):
+        result = access.from_sentry_app(self.out_of_scope_proxy_user, self.out_of_scope_org)
+        assert not result.has_team_access(self.team)
+        assert not result.has_team_access(self.team2)
+        assert not result.has_team_access(self.out_of_scope_team)
+
+    def test_no_access_due_to_invalid_user(self):
```
Comment (scefali) [inline tests/sentry/auth/test_access.py:244]: @dcramer I can confirm this test fails without Armin's fix: https://github.com/getsentry/sentry/pull/19950

**Review by scefali — APPROVED** [review]

File: tests/sentry/auth/test_access.py (hunk 1 above)
Comment (dcramer) [reply tests/sentry/auth/test_access.py:244]: realized i typo'd "confirmed" as "confirm" :)

**Pull request merged**